File: src/main/java/com/ByTech/ByTech/Cart/repositories/ShoppingCartRepository.java
package com.ByTech.ByTech.Cart.repositories;

import com.ByTech.ByTech.Cart.models.ShoppingCart;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
    List<ShoppingCart> findByClient_UserName(String clientEmail);
    ShoppingCart findByClient_UserNameAndProduct_Id(String clientEmail, Long productId);
    ShoppingCart findByProduct_IdAndClient_UserName(Long productId, String clientEmail);
    void deleteByClient_Id(Long clientId);
    Long countByClient_Id(Long clientId);

}

File: Movil/backend/ByTech_movil/src/main/kotlin/com/example/ByTech_movil/ShoppingCart/repositories/ShoppingCartRepository.kt
package com.example.ByTech_movil.ShoppingCart.repositories

import com.example.ByTech_movil.ShoppingCart.models.ShoppingCart
import org.springframework.data.jpa.repository.JpaRepository

interface ShoppingCartRepository : JpaRepository<ShoppingCart?, Long?> {
    fun findByClient_UserName(clientEmail: String?): List<ShoppingCart?>?
    fun findByClient_UserNameAndProduct_Id(clientEmail: String?, productId: Long?): ShoppingCart?
    fun findByProduct_IdAndClient_UserName(productId: Long?, clientEmail: String?): ShoppingCart?
    fun deleteByClient_Id(clientId: Long?)
    fun countByClient_Id(clientId: Long?): Long?
}
